packages/core/src/tracing/instrumentation/cloud/aws-sdk/v3/kinesis.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+'use strict';
+
+const cls = require('../../../../cls');
+const { EXIT } = require('../../../../constants');
+const { getStackTrace } = require('../../../../tracingUtil');
+const { InstanaAWSProduct } = require('./instana_aws_product');
+
+const SPAN_NAME = 'kinesis';
+
+const operationsInfo = {
+  CreateStreamCommand: {
+    op: 'createStream'
+  },
+  DeleteStreamCommand: {
+    op: 'deleteStream'
+  },
+  GetRecordsCommand: {
+    op: 'getRecords'
+  },
+  GetShardIteratorCommand: {
+    op: 'shardIterator'
+  },
+  ListStreamsCommand: {
+    op: 'listStreams'
+  },
+  PutRecordCommand: {
+    op: 'putRecord'
+  },
+  PutRecordsCommand: {
+    op: 'putRecords'
+  },
+  ListShardsCommand: {
+    op: 'listShards'
+  }
+};
+
+const operations = Object.keys(operationsInfo);
+
+class InstanaAWSKinesis extends InstanaAWSProduct {
+  instrumentedSmithySend(ctx, originalSend, smithySendArgs) {
+    // NOTE: `shimSmithySend`  in index.js is already checking the result of `isActive`
+    if (cls.skipExitTracing()) {
+      return originalSend.apply(ctx, smithySendArgs);
+    }
+
+    const command = smithySendArgs[0];
+    return cls.ns.runAndReturn(() => {
+      const self = this;
+      const span = cls.startSpan(this.spanName, EXIT);
+      span.ts = Date.now();
+      span.stack = getStackTrace(this.instrumentedSend, 1);
+      span.data[this.spanName] = this.buildSpanData(command.constructor.name, command.input);
+      this.captureRegion(ctx, span);
+
+      if (typeof smithySendArgs[1] === 'function') {
+        const _callback = smithySendArgs[1];
+
+        smithySendArgs[1] = cls.ns.bind(function (err) {
+          if (err) {
+            _callback.apply(this, arguments);
+            self.finishSpan(err, span);
+          } else {
+            _callback.apply(this, arguments);
+            self.finishSpan(null, span);
```

Comment:
nitpick: I think it doesn't matter much (and I know other services use the same logic), but we usually first send the span and then the customer's fn is called.

Reply:
done 👍